tests/python/test_assert.py

```diff
@@ -39,6 +44,28 @@ def func():
         func()
 
 
+@ti.require(ti.extension.assertion)
+@ti.all_archs_with(debug=True, gdb_trigger=False)
+def test_assert_message_formatted():
+    x = ti.field(dtype=int, shape=16)
+    x[10] = 42
+
+    @ti.kernel
+    def assert_formatted():
+        for i in x:
+            assert x[i] == 0, 'x[%d] expect=%d got=%d' % (i, 0, x[i])
+
+    @ti.kernel
+    def assert_float():
+        y = 0.5
+        assert y < 0, 'y = %f' % y
+
+    with pytest.raises(RuntimeError, match=r'x\[10\] expect=0 got=42'):
+        assert_formatted()
+    with pytest.raises(RuntimeError, match=r'y = 0.5'):
+        assert_float()
```

Comment:
Add an `assert_success` test too?

Reply:
We actually already have
https://github.com/taichi-dev/taichi/blob/c3f1a5d1899500eb08facca78fa1373e279cb692/tests/python/test_assert.py#L42-L50

Here we should simply focus on formatting.

Reply:
Hmm, anyway i flipped `x[0]` back to 0 and run `assert_formatted()` again for the success path.